controlpanel/api/models/parameter.py

```diff
@@ -0,0 +1,74 @@
+from django.core.validators import RegexValidator
+from django.db import models
+from django.db.models.signals import pre_delete, pre_save
+from django.dispatch import receiver
+
+from django_extensions.db.models import TimeStampedModel
+
+from controlpanel.api.aws import aws
+
+
+class Parameter(TimeStampedModel):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._value = None
+
+    @property
+    def value(self):
+        return self._value
+
+    @value.setter
+    def value(self, val):
+        self._value = val
+
+    @property
+    def arn(self):
+        return f"arn:aws:ssm:::parameter{self.name}"
+
+    @property
+    def name(self):
+        return f"/alpha/{self.used_by}/{self.role_name}/secrets/{self.key}"
+
+    key = models.CharField(
+        max_length=50,
+        validators=[RegexValidator(r'[a-zA-Z0-9_]{1,50}')]
+    )
+    description = models.CharField(max_length=600)
+    used_by = models.CharField(
+        max_length=8,
+        choices=(
+            ('airflow', 'Airflow'),
+            ('webapp', 'Web app'),
+        )
+    )
+    role_name = models.CharField(
+        max_length=63,
+        validators=[RegexValidator(r'[a-zA-Z0-9_]{1,63}')]
+    )
+    created_by = models.ForeignKey(
+        "User",
+        on_delete=models.SET_NULL,
+        null=True,
+    )
+
+
+@receiver(pre_save, sender=Parameter)
+def save_aws_parameter(sender, **kwargs):
+    parameter = kwargs['instance']
+    is_create = not parameter.pk
+
+    if is_create:
```

Comment:
What about the update?

Reply:
At the moment there is no update. just create/delete.